File: src/main/java/sage/transfer/FollowInfo.java
package sage.transfer;

import java.util.Collection;

import sage.entity.Follow;

public class FollowInfo {
  private UserLabel user;
  private Collection<TagLabel> tags;
  
  FollowInfo() {}

  public FollowInfo(UserLabel user, Collection<TagLabel> tags) {
    this.user = user;
    this.tags = tags;
  }

  public FollowInfo(Follow follow) {
    user = new UserLabel(follow.getTarget());
    tags = TagLabel.listOf(follow.getTags());
  }
  
  public UserLabel getUser() {
    return user;
  }
  
  public Collection<TagLabel> getTags() {
    return tags;
  }
}

File: src/main/kotlin/sage/transfer/FollowInfo.kt
package sage.transfer

import sage.annotation.KotlinNoArg
import sage.entity.Follow

@KotlinNoArg
class FollowInfo {
  var user: UserLabel? = null
  var tags: Collection<TagLabel> = emptyList()

  constructor(user: UserLabel, tags: Collection<TagLabel>) {
    this.user = user
    this.tags = tags
  }

  constructor(follow: Follow) {
    user = UserLabel(follow.target)
    tags = follow.tags.map { TagLabel(it) }
  }
}
